### PR TITLE
Add default number type configuration

### LINKED ISSUE #1408: Setting Integer as the default number type
## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [ ] feature request
  - [x] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Detailed Description
Hi, is there a way to set integer as the default number type? 90% of the numbers in my app are integers, so I'm forced to do a lot of trickery like this. 

```
type IntOverrideForTsoa = {
    /** @isInt */ id: number,
    /** @isInt */ learning: number,
    /** @isInt */ learned: number,
    /** @isInt */ unseenWords: number,
}
const result: (typeof collection & IntOverrideForTsoa) = {
    ...collection,
    learning: learning.length,
    learned: learned.length,
    unseen: unseenWords.length
}
return result
```

### PR DESCRIPTION
**Pull request by PetrKubes97**
Hi, some time back I've asked about default number types, but unfortunately the issue was closed as stale - https://github.com/lukeautry/tsoa/issues/1408. I went ahead and implemented it anyway as it is super useful for my project. I think it's quite useful, as double as the default number type isn't suitable, if the backend uses for example number ids for entities. The amount of annotation is then just too damn high. 

I wasn't sure how to write proper test for this, as I could find any configuration tests.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

Put `closes #1408`

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

### CODE REVIEW

**Comment by WoH** [comment]
Is there any way to escape from having integers when I have a ts type number?

**Comment by PetrKubes97** [comment]
Not sure what you mean by "escape from having integers when I have a ts type number". When a ts type is number, tsoa defaults to 
```
type: number
format: double
```

Only if I set the default to `integer`, it switches to
```
type: integer,
format: int32,
```
The default behaviour is unchanged.

**Comment by WoH** [comment]
So assume I want to have a double field now: 

```ts
interface T {
  id: number // this is now fine without @isInt
  amount: number // <--- how do I make this a double ?
}

**Comment by PetrKubes97** [comment]
There is a `@isDouble` annotation already, but it is not documented. That might be the source of confusion.

**Comment by WoH** [comment]
Makes sense, I wasn't sure if that could override the integer though.
That would certainly be a useful test then.

**Comment by PetrKubes97** [comment]
@WoH Hi, sorry for the delay. I've added the tests. Had some troubles with different formatting settings, .editorconfig might need some love.

File: packages/cli/src/metadataGeneration/metadataGenerator.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import mm from 'minimatch';
 import { importClassesFromDirectories } from '../utils/importClassesFromDirectories';
 import { ControllerGenerator } from './controllerGenerator';
 import { GenerateMetadataError } from './exceptions';
-import { Tsoa } from '@tsoa/runtime';
+import {Config, Tsoa} from '@tsoa/runtime';
```
Comment (WoH) [inline packages/cli/src/metadataGeneration/metadataGenerator.ts:5]:
```suggestion
import { Config, Tsoa } from '@tsoa/runtime';
```

File: packages/cli/src/module/generate-routes.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { Tsoa } from '@tsoa/runtime';
 import { DefaultRouteGenerator } from '../routeGeneration/defaultRouteGenerator';
 import { fsMkDir } from '../utils/fs';
 import path = require('path');
+import {Config as BaseConfig } from "@tsoa/runtime";
```
Comment (WoH) [inline packages/cli/src/module/generate-routes.ts:8]:
```suggestion
import { Config as BaseConfig } from "@tsoa/runtime";
```

**Review by WoH — APPROVED** [review]

**Pull request merged**